engine/table/src/main/java/io/deephaven/engine/table/impl/by/GroupByChunkedOperator.java

```diff
@@ -331,15 +439,123 @@ public ModifiedColumnSet apply(@NotNull final ModifiedColumnSet upstreamModified
+            try (final ResettableWritableObjectChunk<WritableRowSet, Values> rowSetResettableChunk =
+                         ResettableWritableObjectChunk.makeResettableChunk();
+                 final ResettableWritableObjectChunk<RowSetBuilderRandom, Values> addedBuildersResettableChunk =
+                         ResettableWritableObjectChunk.makeResettableChunk();
+                 final ResettableWritableObjectChunk<RowSetBuilderRandom, Values> removedBuildersResettableChunk =
+                         ResettableWritableObjectChunk.makeResettableChunk();
+                 final RowSequence.Iterator destinationsIterator =
+                         stepDestinations.getRowSequenceIterator()) {
+
+                // noinspection unchecked
+                final WritableObjectChunk<WritableRowSet, Values> rowSetBackingChunk =
+                        rowSetResettableChunk.asWritableObjectChunk();
+                // noinspection unchecked
+                final WritableObjectChunk<RowSetBuilderRandom, Values> addedBuildersBackingChunk =
+                        addedBuildersResettableChunk.asWritableObjectChunk();
+                // noinspection unchecked
+                final WritableObjectChunk<RowSetBuilderRandom, Values> removedBuildersBackingChunk =
+                        removedBuildersResettableChunk.asWritableObjectChunk();
+
+                while (destinationsIterator.hasMore()) {
+                    final long firstSliceDestination = destinationsIterator.peekNextKey();
+                    final long firstBackingChunkDestination =
+                            rowSets.resetWritableChunkToBackingStore(rowSetResettableChunk, firstSliceDestination);
+                    addedBuilders.resetWritableChunkToBackingStore(addedBuildersResettableChunk, firstSliceDestination);
+                    removedBuilders.resetWritableChunkToBackingStore(removedBuildersResettableChunk, firstSliceDestination);
+
+                    final long lastBackingChunkDestination =
+                            firstBackingChunkDestination + rowSetBackingChunk.size() - 1;
+                    final RowSequence initialDestinationsSlice =
+                            destinationsIterator.getNextRowSequenceThrough(lastBackingChunkDestination);
+
+                    initialDestinationsSlice.forAllRowKeys((final long destination) -> {
+                        final int backingChunkOffset =
+                                Math.toIntExact(destination - firstBackingChunkDestination);
+                        try (final WritableRowSet addRowSet =
+                                nullToEmpty(extractAndClearBuilderRandom(addedBuildersBackingChunk, backingChunkOffset));
+                             final WritableRowSet removeRowSet =
+                                nullToEmpty(extractAndClearBuilderRandom(removedBuildersBackingChunk, backingChunkOffset))) {
+                            final WritableRowSet workingRowSet = rowSetForSlot(backingChunkOffset);
```

Comment:
We should consider a change here. If `rowSets.getUnsafe` returns `null`, we can safely just take the `addedRowSet` (and convert it to tracking if needed), rather than having to go through the rigamarole of dealing with reference counting on the `OrderedLongSet`, etc. `removedRowSet` had better be empty in this situation.
Note that a `null` `AutoCloseable` is cool, the try-with-resources will just ignore it.